### ISSUE TITLE
Issues with recent updates in Python schema

### ISSUE THREAD

**Issue by aldanor** [equinusocio/material-theme#436]
I just took the latest master with Python fixes for a spin and noticed a few things that seem off:
- `support.type.python` is still not highlighted unless it's a function call like `int(1)`
- `meta.function-call.arguments.python` is highlighted the same as strings which seems wrong (the variable names are green)
- in fact, it looks like the entire content inside each function call is now highlighted green (see line 4)
- is the slight difference in color between `keyword.operator.logical.python` (`or`, `not`) and `support.function.builtin.python`) (`isinstance`, `issubclass`) intentional? (just wondering)

![image](https://cloud.githubusercontent.com/assets/2418513/9786723/327deca2-57b4-11e5-9ad6-0f558241fe46.png)

Here's Tomorrow 80s:

![image](https://cloud.githubusercontent.com/assets/2418513/9786731/3ecabb5c-57b4-11e5-9585-4a62a29e5bae.png)

Here's github:

```python
hello = 'world'
(x, y, z) = (int, float, dict)
print('{}, {}'.format(hello, 'world'))
TypeError('{} {}'.format(hello, 'world'))
if not isinstance(x, X) or not issubclass(X, Y)
{k: f(hello, 'world', k, v) for k, v in x.items()}
```

Thanks!

**Comment by equinusocio** [comment]
Thanks @aldanor i will check for this.

**Comment by equinusocio** [comment]
This is what you'll see from the next release, make sure you using the latest build:
<img width="427" alt="screen shot 2015-09-19 at 22 16 31" src="https://cloud.githubusercontent.com/assets/10454741/9977983/24372f18-5f1c-11e5-85f3-a2a570c8e196.png">

**Comment by aldanor** [comment]
@equinusocio hmm, this highlighting looks quite wrong to me actually. 

- On line 3, `hello` is an identifier, however it's highlighted the same as literal string; same on lines 4, 5 and 6. It should probably be just white, same as identifiers in other scopes. Or blue (which is what tomorrow 80s does for function arguments).
- On line 4, `.format` is not at all different from `.format` on line 3, but it's highlighted differently. It should probably be blue.
- Built-in types are still not highlighted (`int`, `float`).

**Comment by equinusocio** [comment]
I cant style some elements because the package doesn't give me the scope. I think that the python syntax have some issue with the new system.

**Comment by evenicoulddoit** [comment]
Hey - unfortunately I think the recent changes have actually made things quite significantly worse.

The most obvious one is the inability to distinguish between variables and strings within function calls, below:

![image](https://cloud.githubusercontent.com/assets/3514018/10048928/297d6a02-620d-11e5-9b8a-50ce8e187626.png)

**Comment by htongmt** [comment]
I'm in the same boat with @evenicoulddoit. Had to switch color schemes because it was unbearable. Strings and variable names were all the same color.

**Comment by equinusocio** [comment]
@evenicoulddoit i need some snippet code

**Comment by aldanor** [comment]
I think tomorrow night could be a good reference here in general (re: what should be highlighted and what things can be of the same colour) since it got Python highlighting almost completely right.

**Comment by evenicoulddoit** [comment]
@equinusocio what would you like? The above snippet was:

```python
url(r'^home/$',
    views.CustomerLoginHome.as_view(),
    name='home')
```

**Comment by equinusocio** [comment]
I think this is related with the python syntax because i get one scope for all the "views" string:
<img width="716" alt="screen shot 2015-09-27 at 11 29 03" src="https://cloud.githubusercontent.com/assets/10454741/10122080/19e9b716-650b-11e5-9fdf-12f52305373e.png">

**Comment by evenicoulddoit** [comment]
Unfortunately I don't know enough about the construct of the syntax highlighting to comment, though both strings mentioning "home" should be noticeably different from the `views.CustomerLoginHome.as_view()` method. Its definitely possible because I switched to another scheme which differentiates between the two

**Comment by equinusocio** [comment]
@evenicoulddoit Yes i can make strings different, but the major issue is the "views.Customer... " that i can't fix. What color do you like for strings?

**Comment by evenicoulddoit** [comment]
![image](https://cloud.githubusercontent.com/assets/3514018/10129876/f15cf2f8-65ba-11e5-91c7-0190072d0e45.png)

I think green works well for strings, and perhaps just white for variables used within functions?

**Comment by equinusocio** [comment]
@evenicoulddoit do you like this?

<img width="263" alt="screen shot 2015-09-30 at 19 38 03" src="https://cloud.githubusercontent.com/assets/10454741/10201290/865a1baa-67ab-11e5-8149-ecf7b0316364.png">

**Comment by evenicoulddoit** [comment]
@equinusocio Yeh, that works :)

**Comment by tcwalther** [comment]
@equinusocio Yes, that is much better. +1 on the issue, and also +1 on your latest proposed solution!

**Comment by equinusocio** [comment]
Check the latest release

**Comment by evenicoulddoit** [comment]
I think this is a big improvement for Python:

![image](https://cloud.githubusercontent.com/assets/3514018/10416197/ea54237a-7003-11e5-911e-4e47cf712ed1.png)

But JavaScript seems almost entirely blue:

![image](https://cloud.githubusercontent.com/assets/3514018/10416206/6eab5ce2-7004-11e5-842d-fae25c549b8b.png)

That might be a separate issue, and  I can't remember if its a regression.

**Comment by equinusocio** [comment]
@evenicoulddoit Paste this JS snippet please. What ST build are you using? i remember that from 3085 build sublime text use a new syntax system and this theme support only this new system. There is no legacy support for the oldest syntax system because there is a lot of changes in the native language scopes.

**Comment by Revod** [comment]
Here a pic of my js file, st build 3083
![captura de pantalla de 2015-10-11 10 35 30](https://cloud.githubusercontent.com/assets/10109689/10417408/e6aa051e-7003-11e5-9dea-40a8d85c1665.png)
![captura de pantalla de 2015-10-11 10 37 03](https://cloud.githubusercontent.com/assets/10109689/10417413/098f4198-7004-11e5-9c66-8a2ea2ab830d.png)

**Comment by equinusocio** [comment]
I need a portion of your code snippet

**Comment by Revod** [comment]
```javascript
			// window resize
			$(window).bind('resize.elastislide', function( event ) {
				
				// set values again
				instance._setCurrentValues();
				
				// need to resize items
				if( instance.visibleWidth < instance.options.minItems * ( instance.options.imageW + 2 * instance.options.border ) + ( instance.options.minItems - 1 ) * instance.options.margin ) {
					instance._setDim( ( instance.visibleWidth - ( instance.options.minItems - 1 ) * instance.options.margin ) / instance.options.minItems );
					instance._setCurrentValues();
					instance.fitCount	= instance.options.minItems;
				}	
				else{
					instance._setDim();
					instance._setCurrentValues();
				}
				
				instance.$slider.css({
					width	: instance.sliderW + 10 // TODO: +10px seems to solve a firefox "bug" :S
				});
						
				// slide to the current element
				clearTimeout( instance.resetTimeout );
				instance.resetTimeout	= setTimeout(function() {
					instance._slideToCurrent();
				}, 200);
				
			});
			
			// navigation buttons events
			this.$navNext.bind('click.elastislide', function( event ) {
				instance._slide('right');
			});
			
			this.$navPrev.bind('click.elastislide', function( event ) {
				instance._slide('left');
			});
			
			// item click event
			this.$items.bind('click.elastislide', function( event ) {
				instance.options.onClick( $(this) );
				//return false;
			});
			
			// touch events
			instance.$slider.touchwipe({
				wipeLeft			: function() {
					instance._slide('right');
				},
				wipeRight			: function() {
					instance._slide('left');
				}
			});
			
		}
```

**Comment by equinusocio** [comment]
Ok same thing.
> What ST build are you using? i remember that from 3085 build sublime text use a new syntax system and this theme support only this new system. There is no legacy support for the oldest syntax system because there is a lot of changes in the native language scopes.

<img width="705" alt="screen shot 2015-10-11 at 18 02 12" src="https://cloud.githubusercontent.com/assets/10454741/10417611/8083bd0e-7042-11e5-92e9-d7d0b5f4a950.png">

**Comment by Revod** [comment]
I update sublime text 3 to the dev 3095 and work 100%
![captura de pantalla de 2015-10-11 11 57 45](https://cloud.githubusercontent.com/assets/10109689/10417713/71b4907e-700f-11e5-80a3-71187ecc82dd.png)

**Comment by evenicoulddoit** [comment]
Belle, belle!

![image](https://cloud.githubusercontent.com/assets/3514018/10423188/b96da368-70bb-11e5-85ea-a9885cdfd54f.png)

I understand that its a lot more work to support older versions of Sublime, but bear in mind that 3095 is a *bleeding edge* version, and that its unlikely most will have it. Perhaps you could add a message to `install.txt` so that people won't be surprised when it doesn't look as expected?

**Comment by evenicoulddoit** [comment]
The only criticism I have now with the Python scheme is the lack of distinction between a *function declaration* and a *function invocation*, e.g.:

![image](https://cloud.githubusercontent.com/assets/3514018/10425094/c744fbb0-70cc-11e5-82d1-1b7f8cf009db.png)
 
Here's what Monokai looks like as a reference:
![image](https://cloud.githubusercontent.com/assets/3514018/10425112/e02dcb16-70cc-11e5-9a13-e543bdd35144.png)

**Comment by Fossil01** [comment]
So the JS fixes are present in Develop or also in Master? Cause I just pulled the latest master into ST3 and JS still looks bad.

**Comment by equinusocio** [comment]
@Fossil01 this theme support only the syntax-system used from build 3085+

**Comment by Fossil01** [comment]
Ah that looks much better :-) One more question. The sidebar divider is gone, it now blends into the editor field. Can I add that back in? Same goes for statusbar/searchbar.

**Comment by equinusocio** [comment]
@Fossil01 The sidebar separator never existed :dango:

**Comment by niosus** [comment]
Hm... I guess c++ is still broken, And I guess it is related... Issue #444 (even though it is closed) is not resolved.

**Comment by equinusocio** [comment]
#444 is closed becase is related to #436. And to fix this you have to update ST.
